modules/misc/gtk.nix

```diff
@@ -166,14 +166,26 @@ in {
     home.packages = optionalPackage cfg.font ++ optionalPackage cfg.theme
       ++ optionalPackage cfg.iconTheme;
 
-    home.file.${cfg2.configLocation}.text =
-      concatStringsSep "\n" (mapAttrsToList formatGtk2Option ini) + "\n"
-      + cfg2.extraConfig;
+    home.file.${cfg2.configLocation} = {
+      text = concatStringsSep "\n" (mapAttrsToList formatGtk2Option ini) + "\n"
+        + cfg2.extraConfig;
+      onChange = mkIf (cfg.theme != null)
+        (let xsettingsd = "${pkgs.xsettingsd}/bin/xsettingsd";
+        in ''
+          timeout 2 ${xsettingsd} -c <(echo 'Net/ThemeName "${cfg.theme.name}"') & true
+        '');
+    };
 
     home.sessionVariables.GTK2_RC_FILES = cfg2.configLocation;
 
-    xdg.configFile."gtk-3.0/settings.ini".text =
-      toGtk3Ini { Settings = ini // cfg3.extraConfig; };
+    xdg.configFile."gtk-3.0/settings.ini" = {
+      text = toGtk3Ini { Settings = ini // cfg3.extraConfig; };
+      onChange = mkIf (cfg.theme != null)
+        (let gsettings = "${pkgs.glib}/bin/gsettings";
+        in ''
+          ${gsettings} set org.gnome.desktop.interface gtk-theme ${cfg.theme.name} || true
```

Comment:
The gsettings command doesn't seem to do anything for me. The gtk3 theme switch seems to happen from the xsettingsd call instead.

Reply:
Indeed, attempting to switch theme through dconf also doesn't do anything.

(I don't have gnome-settings-daemon or xsettingsd running)

Reply:
Thanks for the review.

I don't remember exactly which one(s), but i'm pretty sure there were a couple of programs i use that didn't switch with just the xsettingsd call. I _think_ it was firefox.

I'll take a look at it and let you know

Reply:
You're right. I commented it out and everything seems to be working.

I'm gonna go ahead and remove this from the PR for now, and will revert it back if i notice any other programs not switching themes.